api/views/mixins.py

```diff
@@ -0,0 +1,24 @@
+from rest_framework import request, serializers
+from rest_framework.permissions import SAFE_METHODS
+
+
+class ReadWriteSerializersMixin:
+    """
+    A view mixin that provides a way to specify different serializers for
+    input and output purposes.
+    """
+
+    # Typings for attributes which are set in ViewSet
+    request: request.Request
+    serializer_class: type[serializers.BaseSerializer] | None
```

Comment:
Why am I still using `Union` when pipe was already introduced in 3.10? :)

Reply:
<buzzkill mode>Also it was introduced in 3.9, not 3.10</buzzkill mode>

Reply:
I was wrong. Really 3.9 🤦